File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int i=0;
        int j=nums.length-1;
        int index=nums.length-1;
        int ans[]=new int[nums.length];
        //[-7,-3,2,3,11]
        while(i<=j){
            int sum1=nums[i]*nums[i];
            int sum2=nums[j]*nums[j];
            if(sum1>sum2){
                ans[index--]=sum1;
                i++;
            }
            else{
                ans[index--]=sum2;
                j--;
            }

        }
        return ans;
        
    }
}

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: list[int]) -> list[int]:
        i = 0
        j = len(nums) - 1
        ans = [0]*len(nums)
        idx = len(nums)-1
        while i<=j:
            if abs(nums[i]) > abs(nums[j]):
                ans[idx] = nums[i]*nums[i]
                idx-=1
                i+=1
            else:
                ans[idx] = nums[j]*nums[j]
                idx-=1
                j-=1
        return ans

        